tests/Functional/Schema/SqliteSchemaManagerTest.php

```diff
@@ -371,4 +374,57 @@ public function testShorthandInForeignKeyReferenceWithMultipleColumns(): void
             $createTableTrackSql,
         );
     }
+
+    public function testListTableNoSchemaEmulation(): void
+    {
+        $this->dropTableIfExists('`list_table_no_schema_emulation.test`');
+
+        $ddl = <<<'DDL'
+        CREATE TABLE `list_table_no_schema_emulation.test` (
+            id INTEGER,
+            parent_id INTEGER,
+            PRIMARY KEY (id),
+            FOREIGN KEY (parent_id) REFERENCES `list_table_no_schema_emulation.test` (id)
+        );
+        DDL;
+        $dd2 = <<<'DDL'
+        CREATE INDEX i ON `list_table_no_schema_emulation.test` (parent_id);
+        DDL;
+
+        $this->connection->executeStatement($ddl);
+        $this->connection->executeStatement($dd2);
+
+        $databasePlatform = $this->connection->getDatabasePlatform();
+        assert($databasePlatform instanceof SqlitePlatform);
+        $databasePlatform->disableSchemaEmulation();
+
+        $schemaManager = $this->schemaManager;
+        $refl          = new ReflectionMethod($schemaManager, 'selectTableColumns');
+        $refl->setAccessible(true);
```

Comment:
Changing method accessibility in tests is a smell. Why do you need to do it?

Reply:
This is because full table introspection with schema is not supported (yet).

Therefore, I test functional behaviour using protected methods.

Reply:
addressed